opentelekomcloud/resource_opentelekomcloud_cce_node_pool_v3.go

```diff
@@ -0,0 +1,526 @@
+	d.Set("initial_node_count", s.Spec.InitialNodeCount)
+	d.Set("scale_enable", s.Spec.Autoscaling.Enable)
+	d.Set("min_node_count", s.Spec.Autoscaling.MinNodeCount)
+	d.Set("max_node_count", s.Spec.Autoscaling.MaxNodeCount)
+	d.Set("scale_down_cooldown_time", s.Spec.Autoscaling.ScaleDownCooldownTime)
+	d.Set("priority", s.Spec.Autoscaling.Priority)
+
+	labels := map[string]string{}
+	for key, val := range s.Spec.NodeTemplate.K8sTags {
+		if strings.Contains(key, "cce.cloud.com") {
+			continue
+		}
+		labels[key] = val
+	}
+	d.Set("labels", labels)
+
+	var volumes []map[string]interface{}
+	for _, pairObject := range s.Spec.NodeTemplate.DataVolumes {
+		volume := make(map[string]interface{})
+		volume["size"] = pairObject.Size
+		volume["volumetype"] = pairObject.VolumeType
+		volume["extend_param"] = pairObject.ExtendParam
+		volumes = append(volumes, volume)
+	}
+	if err := d.Set("data_volumes", volumes); err != nil {
+		return fmt.Errorf("[DEBUG] Error saving dataVolumes to state for Open Telekom Cloud Node Pool (%s): %s", d.Id(), err)
+	}
+
+	rootVolume := []map[string]interface{}{
+		{
+			"size":         s.Spec.NodeTemplate.RootVolume.Size,
+			"volumetype":   s.Spec.NodeTemplate.RootVolume.VolumeType,
+			"extend_param": s.Spec.NodeTemplate.RootVolume.ExtendParam,
+		},
+	}
+	if err := d.Set("root_volume", rootVolume); err != nil {
+		return fmt.Errorf("[DEBUG] Error saving root Volume to state for Open Telekom Cloud Node Pool (%s): %s", d.Id(), err)
+	}
+
+	d.Set("status", s.Status.Phase)
+
+	return nil
+}
+
+func resourceCCENodePoolUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	nodePoolClient, err := config.cceV3Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating Open Telekom Cloud CCE client: %s", err)
+	}
+
+	initialNodeCount := d.Get("initial_node_count").(int)
+
+	updateOpts := nodepools.UpdateOpts{
+		Kind:       "NodePool",
+		ApiVersion: "v3",
+		Metadata: nodepools.UpdateMetaData{
+			Name: d.Get("name").(string),
+		},
+		Spec: nodepools.UpdateSpec{
+			InitialNodeCount: initialNodeCount,
+			Autoscaling: nodepools.AutoscalingSpec{
+				Enable:                d.Get("scale_enable").(bool),
+				MinNodeCount:          d.Get("min_node_count").(int),
+				MaxNodeCount:          d.Get("max_node_count").(int),
+				ScaleDownCooldownTime: d.Get("scale_down_cooldown_time").(int),
+				Priority:              d.Get("priority").(int),
+			},
+			Type: d.Get("type").(string),
+		},
+	}
+
+	clusterid := d.Get("cluster_id").(string)
+	_, err = nodepools.Update(nodePoolClient, clusterid, d.Id(), updateOpts).Extract()
+	if err != nil {
+		return fmt.Errorf("Error updating Open Telekom Cloud Node Pool: %s", err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"Synchronizing"},
+		Target:     []string{""},
+		Refresh:    waitForCceNodePoolActive(nodePoolClient, clusterid, d.Id()),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
```

Comment:
`Update` timeout should be used in this case ― you need to add `Update` to `Timeouts` and use `schema.TimeoutUpdate` here